### PR TITLE
Fix e2e test by splitting up extensive if condition

### PR DESCRIPTION
**Pull request by waodim**
## Description

[DAQ-3366](https://dt-rnd.atlassian.net/browse/DAQ-3366)

e2e test failed because of the logical OR check in the if condition where we check if logmon is enabled. I removed it but kept the MEID check inside the if condition.

## How can this be tested?

Run the e2e tests

[DAQ-3366]: https://dt-rnd.atlassian.net/browse/DAQ-3366?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by andriisoldatenko — CHANGES_REQUESTED** [review]
it doesn't work,
please make sure your PR passes this e2e test:
```
make test/e2e/applicationmonitoring/metadataenrichment
```

File: pkg/controllers/dynakube/logmonitoring/logmonsettings/reconciler.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 		return nil
 	}
 
-	if !r.dk.LogMonitoring().IsEnabled() || r.dk.Status.KubernetesClusterMEID == "" {
+	if !r.dk.LogMonitoring().IsEnabled() {
 		meta.RemoveStatusCondition(r.dk.Conditions(), conditionType)
 
 		if r.dk.Status.KubernetesClusterMEID == "" {
```
Comment (0sewa0) [inline pkg/controllers/dynakube/logmonitoring/logmonsettings/reconciler.go:43]: you should move the check:
```
		if r.dk.Status.KubernetesClusterMEID == "" {
			return errors.New("the status of the DynaKube is missing information about the kubernetes monitored-entity, skipping LogMonitoring deployment")
		}

```
after the `!r.dk.LogMonitoring().IsEnabled()`

like:
```
if !r.dk.LogMonitoring().IsEnabled() {
	... // has early return
} else if r.dk.Status.KubernetesClusterMEID == "" { 
	return errors.New("the status of the DynaKube is missing information about the kubernetes monitored-entity, skipping LogMonitoring deployment"
}
```

Comment (waodim) [reply pkg/controllers/dynakube/logmonitoring/logmonsettings/reconciler.go:43]: You're right as this should be a separate case. Thanks, will do :)

Comment (waodim) [reply pkg/controllers/dynakube/logmonitoring/logmonsettings/reconciler.go]: done https://github.com/Dynatrace/dynatrace-operator/pull/4205/commits/93f164b0ff1bf116da05052bcc7aebd52c7bb851

@andriisoldatenko please check if the test runs through. Locally for me it ran through successfully several times

**Comment by andriisoldatenko** [comment]
tested in my cluster and it worked!

File: pkg/controllers/dynakube/logmonitoring/logmonsettings/reconciler.go (hunk 1 above)
Comment (andriisoldatenko) [reply pkg/controllers/dynakube/logmonitoring/logmonsettings/reconciler.go]: it works!

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**